tensorboard/data/server/gcs/auth.rs

```diff
@@ -173,6 +174,31 @@ pub enum CredentialsError {
     Unsupported(PathBuf, UnsupportedCredentialsError),
 }
 
+const SCOPES: &[&str] = &["https://www.googleapis.com/auth/cloud-platform"];
+
+/// Attempts to retrieve a service `gcp_auth::Token`, and convert it into a consistent format
+fn gce_service_auth_token() -> Result<BoundedToken, gcp_auth::Error> {
+    async fn service_account_token() -> Result<gcp_auth::Token, gcp_auth::Error> {
+        let authentication_manager = gcp_auth::AuthenticationManager::new().await?;
+        let token_res = authentication_manager.get_token(SCOPES).await;
+        token_res
+    }
+
+    let service_token = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(service_account_token())?;
+
+    let serialized = serde_json::to_string(&service_token).unwrap();
+    let deserialized: OauthTokenResponse = serde_json::from_str(&serialized).unwrap();
```

Comment:
This is hacky, I am more than happy to hear how to rewrite this.

Essentially, we need to create a `BoundedToken` with an `AccessToken` and a time `Instant`.
- I am not sure what is the best way to create the `AccessToken` given that it is deemed "private" in the code.
- The `gcp_auth::Token` has an optional `expires_at: Option<OffsetDateTime>` attribute which is effectively the `BoundedToken.expires` that we need. I am not sure how to convert that to the expected format.